src/main/java/com/devicehive/application/hazelcast/CuratorConfiguration.java

```diff
@@ -34,14 +38,12 @@ CuratorFramework curatorFramework(@Value("${zookeeper.connect}") String zooKeepe
     }
 
     @Bean
-    ServiceInstance<Void> serviceInstance(@Value("${hazelcast.port}") int port, ApplicationContext applicationContext,
-                                          CuratorFramework curatorFramework) throws Exception {
-        final String hostName = InetAddress.getLocalHost().getHostName();
+    ServiceInstance<Void> serviceInstance(@Value("${hazelcast.port}") int port) throws Exception {
         return ServiceInstance
                 .<Void>builder()
-                .name(applicationContext.getId())
+                .name(env.getProperty("curator.service"))
                 .uriSpec(new UriSpec("{address}:{port}"))
-                .address(hostName)
+                .address(InetAddress.getLocalHost().getHostName())
```

Comment:
Yulya, could you please double check if this would cause problems when app deployed in docker/virtual machine?